include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -435,71 +435,28 @@ struct __sycl_usm_alloc
     }
 };
 
-//-----------------------------------------------------------------------
-// type traits for objects granting access to some value objects
-//-----------------------------------------------------------------------
-
-template <typename _ContainerOrIterator>
-struct __memobj_traits
-{
-    using value_type = typename _ContainerOrIterator::value_type;
-};
-
 template <typename _T>
-struct __memobj_traits<_T*>
-{
-    using value_type = _T;
-};
-
-} // namespace __internal
-
-template <typename _ExecutionPolicy, typename _T>
-using __buffer = __internal::__buffer_impl<::std::decay_t<_ExecutionPolicy>, _T>;
-
-template <typename T>
-struct __repacked_tuple
-{
-    using type = T;
-};
-
-template <typename... Args>
-struct __repacked_tuple<::std::tuple<Args...>>
-{
-    using type = oneapi::dpl::__internal::tuple<Args...>;
-};
-
-template <typename T>
-using __repacked_tuple_t = typename __repacked_tuple<T>::type;
-
-template <typename _ContainerOrIterable>
-using __value_t = typename __internal::__memobj_traits<_ContainerOrIterable>::value_type;
-
-template <typename _T>
-struct __accessor
+struct __accessor_impl
```

Comment:
Looks like it's not required to change the name of this class: we doesn't use in code `std::decay` for params of this class and doesn't create any alias for them. So I preferer to have old class name.

Reply:
I've reverted these changes.